applications/ContactStructuralMechanicsApplication/custom_processes/normal_gap_process.cpp

```diff
@@ -0,0 +1,131 @@
+// KRATOS  ___|  |                   |                   |
+//       \___ \  __|  __| |   |  __| __| |   |  __| _` | |
+//             | |   |    |   | (    |   |   | |   (   | |
+//       _____/ \__|_|   \__,_|\___|\__|\__,_|_|  \__,_|_| MECHANICS
+//
+//  License:		 BSD License
+//					 license: StructuralMechanicsApplication/license.txt
+//
+//  Main authors:    Vicente Mataix Ferrandiz
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "custom_processes/normal_gap_process.h"
+#include "contact_structural_mechanics_application_variables.h"
+
+namespace Kratos
+{
+template<SizeType TDim, SizeType TNumNodes, SizeType TNumNodesMaster>
+void NormalGapProcess<TDim, TNumNodes, TNumNodesMaster>::Execute()
+{
+    KRATOS_TRY
+
+    // We get the process info
+    const ProcessInfo& r_process_info = mrMasterModelPart.GetRootModelPart().GetProcessInfo();
```

Comment:
usually all submodelparts have the same processinfo

Reply:
I have trust issues...imagine we replace that in the future...